include/ttmlir/Dialect/TTNN/IR/TTNNOps.td

```diff
@@ -547,4 +547,36 @@ def TTNN_DeallocOp : TTNN_Op<"dealloc"> {
     let arguments = (ins AnyRankedTensor:$input);
 }
 
+def TTNN_AllGatherOp: TTNN_Op<"all_gather"> {
+    let summary = "All gather op.";
+    let description = [{
+        Tensor All Gather operation
+    }];
+
+    let arguments = (ins AnyRankedTensor:$input,
+                         I32Attr:$dim,
+                         I32Attr:$num_links);
```

Comment:
https://github.com/tenstorrent/tt-metal/blob/main/ttnn/cpp/ttnn/operations/ccl/all_gather/all_gather.hpp

From TTNN implementation num_links is an optional argument with default value 1. We should model this in our TTNN IR as well using DefaultValuedAttr:
https://mlir.llvm.org/docs/DefiningDialects/Operations/#attributes-with-default-values

Reply:
Good suggestion! I'm surprised `DefaultValuedAttr` isn't used by any other op.